### PR TITLE
Fail CI check if 'record_soft_failure' without valid reference

### PR DESCRIPTION
**Pull request by rfan1**
poo#115343

- Related ticket: https://progress.opensuse.org/issues/115343
- Needles: n/a
- Verification run: 
https://openqa.suse.de/tests/9627721
https://openqa.opensuse.org/tests/2762755

### CODE REVIEW

**Review by foursixnine — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -126,6 +126,16 @@ if (match_has_tag('yast2_missing_package')) {
...
 }
 ```
+* Please add a bug/ticket reference for `record_soft_failure`, otherwise
+  CI checks may fail. you can use formats like below:
+  bsc#12345 -> Bugzilla bug
+  poo#12345 -> Progress ticket
+  jsc#SLE-19640 -> Jira ticket
+  Maniphest#T5531
+  fate.suse.com/123
+  If you don't have a reference ticket, and still  want to mark a specific
```
Comment (okurz) [inline CONTRIBUTING.md:136]: s/still  want/still want/

Comment (rfan1) [reply CONTRIBUTING.md:136]: Thanks @okurz

File: lib/trento.pm [hunk 2]
```diff
@@ -602,7 +602,7 @@ sub cypress_exec {
         $self->result("fail");
     }
     if ($failok) {
-        record_soft_failure("Cypress exit code:$ret at $log_prefix") if ($ret);
+        record_info('Softfail', "Cypress exit code:$ret at $log_prefix", result => 'softfail') if ($ret);
```
Comment (mpagot) [inline lib/trento.pm:605]: ok

File: tests/sles4sap/sapconf.pm [hunk 3]
```diff
@@ -58,7 +58,7 @@ sub run_developers_tests {
         # Skip results of fate#325548 on SLES4SAP versions before 15
         next if ($summary =~ /fate325548/ and is_sle('<15'));
         if ($summary =~ /Test #(bsc|fate)([0-9]+)/) {
-            record_soft_failure "$1#$2";
+            record_info('Softfail', "$1#$2", result => 'softfail');
```
Comment (alvarocarvajald) [inline tests/sles4sap/sapconf.pm:61]: We need to use `record_soft_failure` here to mark the whole job as soft failed and `record_info` will not do that. And as can be seen from `$1#$2` a `bsc#` and/or `fate#` was included in the message as expected by the CI checks. Is there a way to include this again?

File: tests/sles4sap/sys_param_check.pm [hunk 4]
```diff
@@ -36,7 +36,7 @@ sub add_softfail {
     my ($module, $os_version, $bsc_number, @parameters) = @_;
     foreach my $parameter (@parameters) {
         if (check_var("VERSION", $os_version) && check_failure($module, $parameter)) {
-            record_soft_failure "$bsc_number - Wrong value for $parameter";
+            record_info('Softfail', "$bsc_number - Wrong value for $parameter", result => 'softfail');
```
Comment (alvarocarvajald) [inline tests/sles4sap/sys_param_check.pm:39]: We need to use `record_soft_failure` here to mark the whole job as soft failed and `record_info` will not do that. As can be seen, the message to `record_soft_failure` does include a bsc# (via a variable). Is there a way to include this again?

Comment (rfan1) [reply tests/sles4sap/sys_param_check.pm:39]: hmm, I can add exception for this dedicated string "bsc_number"

File: tests/sles4sap/sapconf.pm (hunk 3 above)
Comment (rfan1) [reply tests/sles4sap/sapconf.pm:61]: Let me check with @foursixnine to find a solution.

Comment (rfan1) [reply tests/sles4sap/sapconf.pm:61]: Thank you @alvarocarvajald,
Let me file a new PR as your comment.

Comment (rfan1) [reply tests/sles4sap/sapconf.pm:61]: I will do that tomorrow :)

Comment (foursixnine) [reply tests/sles4sap/sapconf.pm:61]: @alvarocarvajald would you be ok if we call it `reference` or `issue` instead of `bsc_number`?

Comment (alvarocarvajald) [reply tests/sles4sap/sapconf.pm:61]: Sure, that sounds OK as well.

Comment (rfan1) [reply tests/sles4sap/sapconf.pm:61]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15676
@foursixnine @alvarocarvajald

**Comment by pevik** [comment]
@rfan1 @foursixnine CI should either do eval or ignore cases where variable is used. See example of workaround to make CI happy without actual check :( https://github.com/pevik/os-autoinst-distri-opensuse/commit/70178bffd682b6d83679c09166f1b7c64c640d36

**Pull request merged**